src/server/cmd/worker/main.go

```diff
@@ -169,8 +169,8 @@ func do(appEnvObj interface{}) error {
 
 	// Prepare to write "key" into etcd by creating lease -- if worker dies, our
 	// IP will be removed from etcd
-	ctx, _ := context.WithTimeout(context.Background(), 30*time.Second)
-	resp, err := etcdClient.Grant(ctx, 60 /* seconds */)
+	ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
+	resp, err := etcdClient.Grant(ctx, 10 /* seconds */)
```

Comment:
Yeah that's basically it.  Technically having the old IPs around shouldn't stop new jobs from completing.  We'd just have a couple zombie worker goros on the server side that get shut down once the old IPs expire.